### DISCUSSION TITLE
给<ShareElement>组件添加的transform和key属性消失

### DISCUSSION THREAD

**Discussion by MaxtuneLee** [NervJS/taro#13298] [Q & A] [answered]
我使用的是Taro 3.5.10 React
下面的ShareElement中我已经添加了 transform 和 key 属性
```jsx
<ShareElement key="card" transform duration={300} className="share-card">
          <Paper
            className="card"
            onClick={() => {
              showPage("打卡详情");
            }}
          >
            <View className="title">打卡详情</View>
            <View className="content-body">
              <View className="content">{signData.date}</View>
              <View className="sub-content">{signData.site}</View>
            </View>
          </Paper>
</ShareElement>
```
然而在微信开发者工具中查看时 transform 和 key 属性消失了
```html
<share-element bindtap="eh" data-sid="_Ap" duration="300" easing-function="ease-out" id="_Ap" class="share-card">
    …………
</share-element>
```
请问有大佬知道怎么回事吗，help！

**Comment by tourze** [comment]
貌似有个特性，对于一些非标准的组件属性会在实际渲染时去掉。具体可以翻翻文档。

**Comment by TheKonka** [comment, marked as answer]
`key` 是 `React` 保留属性，所以虽然你添加了 `key`，但最终属性消失
针对此问题，`taro`采用的方法是用其他字段代替，目前可以用`mapkey` 代替 `key`，`taro`的插件会自动替换模版的`mapkey`属性 为 `key`
https://github.com/NervJS/taro/blob/83ce87d3b2d3e74df16ceb7cafe90ffb836bf1d2/packages/taro-weapp/src/template.ts#L35-L38
![image](https://user-images.githubusercontent.com/22173084/218916394-408ba7de-ff08-4b87-be60-04937f5ddca6.png)
![image](https://user-images.githubusercontent.com/22173084/218917004-b3e2f308-17a7-41f0-9ae2-770d086fe6fc.png)
![image](https://user-images.githubusercontent.com/22173084/218917087-408266f6-5764-4c07-80ed-5c61b68d9c36.png)
至于`transform` 属性消失，我看原生小程序也有这个问题，暂不清楚原因
